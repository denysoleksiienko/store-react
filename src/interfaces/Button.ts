export interface IButtonProps {
  title: string;
  type: string;
}
