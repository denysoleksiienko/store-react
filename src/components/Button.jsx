import { StyledButton } from 'styled';

export const Button = ({ title, onClick }) => (
  <StyledButton variant="primary" size="lg" onClick={onClick} type="submit">
    {title}
  </StyledButton>
);
