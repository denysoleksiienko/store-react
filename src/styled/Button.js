import styled from 'styled-components';
import Button from 'react-bootstrap/Button';
import { COLORS } from '../constants/colors';

export const StyledButton = styled(Button)`
  background-color: ${COLORS.orchid};
  width: 100%;
`;
