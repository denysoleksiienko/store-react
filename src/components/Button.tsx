import React from 'react';
import { StyledButton } from 'styled';

import { IButtonProps } from 'interfaces/Button';

export const Button: React.FC<IButtonProps> = ({ title, type }) => (
  <StyledButton variant="primary" size="lg" type={type}>
    {title}
  </StyledButton>
);
