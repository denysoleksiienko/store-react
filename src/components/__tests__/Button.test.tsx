import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { Button } from '../Button';

const props = {
  title: 'Continue',
  type: 'submit',
};

describe('Button component', () => {
  it('should be defined', () => {
    const wrapper = shallow(<Button {...props} />);
    expect(wrapper).toBeDefined();
  });

  it('should match snapshot', () => {
    const wrapper = renderer.create(<Button {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
