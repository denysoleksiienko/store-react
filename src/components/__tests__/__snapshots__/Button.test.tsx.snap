// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<button
  className="sc-gKsewC jrKcHd btn btn-primary btn-lg"
  disabled={false}
  type="submit"
>
  Continue
</button>
`;
